TestForTooMuchCloning.test fails

### Description

I have seen this fail in periodic CI builds and on some PR builds.

```
TestForTooMuchCloning > test FAILED
--
  | java.lang.AssertionError: too many calls to IndexInput.clone during merging: 519
  | at __randomizedtesting.SeedInfo.seed([274680B90E738EFA:AF12BF63A08FE302]:0)
  | at org.junit.Assert.fail(Assert.java:89)
  | at org.junit.Assert.assertTrue(Assert.java:42)
  | at org.apache.lucene.index.TestForTooMuchCloning.test(TestForTooMuchCloning.java:63)
  | at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  | at java.base/java.lang.reflect.Method.invoke(Method.java:580)

```

Or:

```
TestForTooMuchCloning > test FAILED
    java.lang.AssertionError: too many calls to IndexInput.clone during merging: 503
        at __randomizedtesting.SeedInfo.seed([FB5D79942E80A9F0:7309464E807CC408]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.apache.lucene.index.TestForTooMuchCloning.test(TestForTooMuchCloning.java:63)
```

### Gradle command to reproduce

```
./gradlew test --tests TestForTooMuchCloning.test -Dtests.seed=FB5D79942E80A9F0 -Dtests.locale=sl-Latn-SI -Dtests.timezone=America/Mexico_City -Dtests.asserts=true -Dtests.file.encoding=UTF-8
```

--- a/lucene/core/src/test/org/apache/lucene/index/TestForTooMuchCloning.java
+++ b/lucene/core/src/test/org/apache/lucene/index/TestForTooMuchCloning.java
@@ -42,6 +42,8 @@ public class TestForTooMuchCloning extends LuceneTestCase {
             random(),
             dir,
             newIndexWriterConfig(new MockAnalyzer(random()))
+                // to reduce flakiness on merge clone count
+                .setMergeScheduler(new SerialMergeScheduler())
                 .setMaxBufferedDocs(2)
                 // use a FilterMP otherwise RIW will randomly reconfigure
                 // the MP while the test runs
@@ -62,7 +64,7 @@ public class TestForTooMuchCloning extends LuceneTestCase {
     // System.out.println("merge clone count=" + cloneCount);
     assertTrue(
         "too many calls to IndexInput.clone during merging: " + dir.getInputCloneCount(),
-        dir.getInputCloneCount() < 500);
+        dir.getInputCloneCount() < 600);
 
     final IndexSearcher s = newSearcher(r);
     // important: set this after newSearcher, it might have run checkindex

